app/models/components/course/experience_points_records_ability_component.rb

```diff
@@ -0,0 +1,49 @@
+module Course::ExperiencePointsRecordsAbilityComponent
+  include AbilityHost::Component
+
+  def define_permissions
+    # cancancan is unable to merge abilities if administrators are not excluded
```

Comment:
I would very much prefer not using blocks with scopes if possible. This makes the authorisation behaviour inconsistent.